lib/engine/game/g_18_india/step/certificate_selection.rb

```diff
@@ -11,13 +11,18 @@ module Game
     module G18India
       module Step
         class CertificateSelection < Engine::Step::Base
-          attr_reader :companies, :choices
+          attr_reader :choices
 
-          ACTIONS = %w[bid].freeze
-          ACTIONS_WITH_PASS = %w[bid pass].freeze
+          ACTIONS = %w[select_multiple_companies].freeze
+
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] if finished?
+
+            ACTIONS
+          end
 
           def setup
-            @companies = @game.companies
             @choices = Hash.new { |h, k| h[k] = [] }
```

Comment:
You're setting up a hash whose keys are players and values are arrays of companies. Will this ever have any more than a single player in the hash? Or will it just be the current player?
If it's just the current player, then this can be changed to be a simple array.
Or you could have it as a hash whose values are companies and values are booleans.